### PR TITLE
Test Preview for routing forms

### LINKED ISSUE #5413: [CAL-329] Testing Routing forms
Ability to fully test form, router, etc without committing to database

> From [SyncLinear.com](https://synclinear.com)

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Adds a Test Preview to test routing forms without submitting any data to the database.

Fixes #5413

 Loom Video: https://www.loom.com/share/fa517cad4fdd4b80b22befb152b6a3b3

**Environment**: Staging(main branch)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: packages/app-store/ee/routing-forms/components/SingleForm.tsx [hunk 1]
```diff
@@ -210,76 +225,151 @@ function SingleForm({ form, appUrl, Page }: SingleFormComponentProps) {
...
+                <div className="w-full rounded-md border border-gray-200 p-8">
+                  <RoutingNavBar appUrl={appUrl} form={form} />
+                  <Page hookForm={hookForm} form={form} appUrl={appUrl} />
+                </div>
               </div>
             </div>
+          </ShellMain>
+        </FormActionsProvider>
+      </Form>
+      <Dialog open={isTestPreviewOpen} onOpenChange={setIsTestPreviewOpen}>
```
Comment (CarinaWolli) [inline packages/app-store/ee/routing-forms/components/SingleForm.tsx:304]: New dialog to test routing form

File: packages/app-store/ee/routing-forms/pages/routing-link/[...appPages].tsx [hunk 2]
```diff
@@ -124,57 +122,7 @@ function RoutingForm({ form, profile, ...restProps }: inferSSRProps<typeof getSe
                         </p>
                       ) : null}
                     </div>
-                    {form.fields?.map((field) => {
```
Comment (CarinaWolli) [inline packages/app-store/ee/routing-forms/pages/routing-link/[...appPages].tsx:127]: Copied this code into the new `FormInputFields` Component to reuse the code

**Comment by hariombalhara** [comment]
Awesome !!

Let's add tests also in the meantime.

File: packages/app-store/ee/routing-forms/components/SingleForm.tsx [hunk 3]
```diff
@@ -1,11 +1,13 @@
 import { App_RoutingForms_Form } from "@prisma/client";
-import { useEffect } from "react";
+import Link from "next/link";
```
Comment (hariombalhara) [inline packages/app-store/ee/routing-forms/components/SingleForm.tsx:2]: unused import

File: packages/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets.tsx [hunk 4]
```diff
@@ -104,7 +104,6 @@ const MultiSelectWidget = ({
   return (
     <Select
       className="dark:border-darkgray-300 mb-2 block w-full min-w-0 flex-1 rounded-none rounded-r-sm border-gray-300 dark:bg-transparent dark:text-white dark:selection:bg-green-500 disabled:dark:text-gray-500 sm:text-sm"
-      menuPosition="fixed"
```
Comment (hariombalhara) [inline packages/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets.tsx:107]: Why are we removing this?

**Review by hariombalhara — COMMENTED** [review]
Code looks good to me. Will test soon.

**Comment by hariombalhara** [comment]
Screenshot 2022-11-09 at 4 37 17 PM
Right now Dialog isn't scrollable

File: packages/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets.tsx (hunk 4 above)
Comment (CarinaWolli) [reply packages/app-store/ee/routing-forms/components/react-awesome-query-builder/widgets.tsx:107]: The options from the select did show up at a random place in the new Dialog (not right under it). I tested the live form and it didn't change anything there. @hariombalhara Is this needed for anything that I am not aware of?

**Comment by CarinaWolli** [comment]
> Right now Dialog isn't scrollable

That's weird, it is scrollable for me. @hariombalhara What browser are you using? I tested it in chrome and safari and it worked fine as shown in the loom video

**Comment by hariombalhara** [comment]
> That's weird, it is scrollable for me. @hariombalhara What browser are you using? I tested it in chrome and safari and it worked fine as shown in the loom video

I am using Chrome only. Will retest and share a birdeatsbug.

**Comment by hariombalhara** [comment]
> I am using Chrome only. Will retest and share a birdeatsbug.

This issue doesn't happen now. Maybe I was on an older commit.

But found a different problem, it doesn't look good on mobile. https://app.birdeatsbug.com/workspaces/62c323a9-51c4-4184-84ab-72dd1c5f8def/collections/d323e796-f4b1-421b-a184-03b60216357d/sessions/rWShW-3lrSEskib6yBkktPvoAQ_32u8pk3Mo7RK-S_gb

**Comment by CarinaWolli** [comment]
> But found a different problem, it doesn't look good on mobile.

Fixed the mobile view 👍

**Review by hariombalhara — APPROVED** [review]

**Comment by hariombalhara** [comment]
@CarinaWolli Approved and enabled auto-merge.
Let's merge the tests in a follow-up PR.

**Pull request merged**